compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/generators/DataClassMembersGenerator.kt

```diff
@@ -191,18 +191,21 @@ class DataClassMembersGenerator(val components: Fir2IrComponents) {
                 } else
                     null
             }
+            val scope = klass.unsubstitutedScope(
+                components.session,
+                components.scopeSession,
+                withForcedTypeCalculator = true
+            )
             val contributedFunctionsInSupertypes =
                 @OptIn(ExperimentalStdlibApi::class)
                 buildMap<Name, FirSimpleFunction> {
                     for (name in listOf(EQUALS, HASHCODE_NAME, TO_STRING)) {
-                        klass.unsubstitutedScope(
-                            components.session,
-                            components.scopeSession,
-                            withForcedTypeCalculator = true
-                        ).processFunctionsByName(name) {
+                        scope.processFunctionsByName(name) {
                             val declaration = it.fir
                             if (declaration.matchesDataClassSyntheticMemberSignatures) {
-                                putIfAbsent(declaration.name, declaration)
+                                if (putIfAbsent(declaration.name, declaration) == null) {
+                                    return@processFunctionsByName
```

Comment:
This does not stop `processFunctionsByName`. Some time ago we had an opportunity to return `STOP` from this lambda and stop following functions processing. Now it's eliminated (may be we should return to this idea if it provides significant performance boost).

Reply:
I remember seeing that, and I actually expect that effect. :(